File: factory-method/c#/Meats/MysteryMeat.cs
using System;

namespace FactoryMethod
{
    public class MysteryMeat : ICookedMeat
    {
        public decimal Weight { get; set; }
        public string Flavouring { get; set; }

        public void Eat()
        {
            Console.WriteLine("Hmm... suspicious flavour... like cat food or siu mai...");
        }
    }
}

File: factory-method/typescript/src/Meats/MysteryMeat.ts
import {ICookedMeat} from "Interfaces/ICookedMeat";
export class MysteryMeat implements ICookedMeat{
  public Weight: number;
  public Flavouring: string;
  public Eat():void{
    console.log("Hmm... suspicious flavour... like cat food or siu mai...");
  }
}